arangod/Cluster/TakeoverShardLeadership.cpp

```diff
@@ -276,20 +277,21 @@ bool TakeoverShardLeadership::first() {
       error << "TakeoverShardLeadership: failed to lookup local collection "
             << shard << "in database " + database;
       LOG_TOPIC("65342", ERR, Logger::MAINTENANCE) << error.str();
-      _result = actionError(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND, error.str());
+      res = actionError(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND, error.str());
+      result(res);
     }
   } catch (std::exception const& e) {
     std::stringstream error;
 
     error << "action " << _description << " failed with exception " << e.what();
     LOG_TOPIC("79443", WARN, Logger::MAINTENANCE)
         << "TakeoverShardLeadership: " << error.str();
-    _result.reset(TRI_ERROR_INTERNAL, error.str());
+    result(TRI_ERROR_INTERNAL, error.str());
   }
 
-  if (_result.fail()) {
+  if (res.fail()) {
```

Comment:
This would now miss the error set in the exception handler.

Reply:
correct